mantidproject/mantid
# Find conda debug issue by passing debug flags into make CMake presets

**Description of work.**
Conda completely ignores the intension to build with minimal optimisations, instead appending it's own CXXFLAGS. These CXX flags are put into the CMAKE_CXX_FLAGS variable. The debug flags are then appended on, which is typically just appending a "-g" to include the debug symbols. Because the default turns -O2 on we need to overwrite this in the debug flags by passing in -O0. 

I feel this is a better approach than putting compiler specific handling into our CMake files. 



**To test:**

1. cmake .. --preset=linux -DCMAKE_VERBOSE_MAKEFILE=ON
2. ninja -v
3. You could also test debugging some stuff. There shouldn't be any missing or optimised out code.

You should be able to now view the exact compile commands. You'll notice a -O2 followed by a -O0, which overrides the previous setting. 



Fixes #32786. 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
